include/model/CModelFactory.h

```diff
@@ -456,6 +458,17 @@ class MODEL_EXPORT CModelFactory {
     //! A cache of influence calculators for collections of features.
     mutable TStrFeatureVecPrInfluenceCalculatorCPtrMap m_InfluenceCalculatorCache;
 };
+
+class CModelAllocator : public maths::common::CModelAllocator {
+public:
+    explicit CModelAllocator(CResourceMonitor& resourceMonitor)
+        : m_ResourceMonitor{&resourceMonitor} {}
+
+    bool areAllocationsAllowed() const override;
+
+private:
+    CResourceMonitor* m_ResourceMonitor;
+};
```

Comment:
Again I'd suggest something like `CModelMemoryCircuitBreaker`. Also I wonder if this more naturally belongs in `CAnomalyDetectorModel.h`.

Reply:
Moved 👍